airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/source.py

```diff
@@ -81,7 +81,7 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
 
         found_a_file = False
         try:
-            for filepath in self.stream_class.filepath_iterator(logger, config.get("provider")):
+            for filepath in self.stream_class(**config).filepath_iterator():
```

Comment:
First I've made it non static cause I was storing endpoint not in "provider"  but in separate "server" field and it was pain to pass all the configs first to the filepath_iterator and then to _list_backet static functions instead of just passing self. Than I merged "server" field into "provider" and decided to keep filepath_iterator function as instance method  in case we would want to pass additional parameters to those functions in the future.